pghoard/rohmu/compressor.py

```diff
@@ -114,23 +114,32 @@ def decompress_from_fileobj_to_fileobj(self, fsrc, metadata, rsa_private_key=Non
             fsrc = SnappyFile(fsrc)  # pylint: disable=redefined-variable-type
         return fsrc
 
-    def compress_filepath(self, filepath, targetfilepath, compression_algorithm, rsa_public_key=None):
+    def compress_filepath(self, filepath=None, targetfilepath=None,
+                          compression_algorithm=None, rsa_public_key=None, fileobj=None):
```

Comment:
This function implicitly closes any `fileobj` passed in by the caller.  I guess it works here, but it's a bit weird API.